### PR TITLE
Add sanity check to RingMap

### LINKED ISSUE #270: Unexpected result when RingMap is used for rings in different instances of a CAS
### Steps to reproduce
```gap
gap> LoadPackage( "RingsForHomalg" );
gap> QQ1 := HomalgFieldOfRationalsInSingular();
gap> QQ2 := HomalgFieldOfRationalsInSingular();
gap> R := QQ1*"x";
Q[x]
gap> S := QQ2*"a";
Q[a]
gap> m := RingMap( HomalgMatrix("[a]",2,1,S), R, S );
<A "homomorphism" of rings>
gap> Display( Pullback( m, HomalgMatrix( "[[x]]", 1, 1, R ) ) );
a,
0 
gap> Display( Pullback( m, HomalgMatrix( "[[x]]", 1, 1, R ) ) );
a
gap> Display( Pullback( m, HomalgMatrix( "[[x]]", 1, 1, R ) ) );
0
gap> Display( Pullback( m, HomalgMatrix( "[[x]]", 1, 1, R ) ) );
Error, the external CAS Singular (running with PID 22234) returned the following error:
   ? can not convert map(_) -> matrix
   ? error occurred in or before STDIN line 1358: `matrix homalg_variable_17 = homalg_variable_13(homalg_variable_13);`
   ? expected matrix-expression. type 'help matrix;'
```

### Expected result
Either `Display( Pullback( m, HomalgMatrix( "[[x]]", 1, 1, R ) ) );` always returns `a`, or an error is thrown indicating that ring maps between polynomial rings over non-identical coefficient rings are not supported.

### Actual result
The inconsistencies seen above occur.

### Note
If polynomial rings over non-identical coefficient rings are not supported, I can try to create a PR adding an appropriate error.

### PR DESCRIPTION
**Pull request by zickgraf**
All other installations of RingMap seem to delegate to this installation.

Fixes #270.

### CODE REVIEW

**Comment by zickgraf** [comment]
I have force-pushed a check using `IsHomalgExternalRingRep`.

**Comment by zickgraf** [comment]
This does not work: `RingsForHomalg` is loaded after `MatricesForHomalg`, so `IsHomalgExternalRingRep` is not available while loading `MatricesForHomalg`. Should I simply check if `IsBound( S!.ring!.stream )`? If yes, can I be sure that `S!.ring` is always bound or should I check for this, too.

**Comment by mohamed-barakat** [comment]
Ping :)

File: MatricesForHomalg/gap/HomalgRingMap.gi [hunk 1]
```diff
@@ -127,6 +127,14 @@ InstallMethod( RingMap,
   function( images, S, T )
     local c, map, type;
     
+    if IsHomalgExternalRingRep( S ) <> IsHomalgExternalRingRep( T ) then
```
Comment (mohamed-barakat) [inline MatricesForHomalg/gap/HomalgRingMap.gi]:
```suggestion
    if IsHomalgInternalRingRep( S ) <> IsHomalgInternalRingRep( T ) then
```

File: MatricesForHomalg/gap/HomalgRingMap.gi [hunk 2]
```diff
@@ -127,6 +127,14 @@ InstallMethod( RingMap,
...
+        Error( "the source ring and the target ring must either both be an external or both be an internal ring" );
+    fi;
+    
+    if IsHomalgExternalRingRep( S ) and not IsIdenticalObj( S!.ring!.stream, T!.ring!.stream ) then
```
Comment (mohamed-barakat) [inline MatricesForHomalg/gap/HomalgRingMap.gi]:
```suggestion
    if not IsHomalgInternalRingRep( S ) and not IsIdenticalObj( S!.ring!.stream, T!.ring!.stream ) then
```

**Comment by mohamed-barakat** [comment]
My suggested changes lets `MatricesForHomalg` load properly, but other tests still fail, since a non-internal ring is not necessarily and external ring. We can discuss this tomorrow.

**Comment by zickgraf** [comment]
I have force-pushed a new version and the CI is green :-)

**Pull request merged**